radar-upload-backend/src/main/java/org/radarbase/upload/inject/DoaEntityManagerFactory.kt

```diff
@@ -40,8 +33,10 @@ fun <T> EntityManager.transact(transaction: EntityManager.() -> T): T {
         try {
             getTransaction().commit()
         } catch (exe: Exception) {
-            logger.warn("Rolling back operation: {}", exe.toString())
-            getTransaction().rollback()
+            logger.error("Rolling back operation: {}", exe)
+            if(getTransaction() != null && getTransaction().isActive) {
```

Comment:
Would it make sense to use a single `val transaction = getTransaction()` variable and use that throughout this code block? The null-check would need to happen only once, and here we can just check whether the transaction is active.

Reply:
Yep. Agreed